### ISSUE TITLE
PowershellFileTask generates invalid yaml

### ISSUE THREAD

**Issue by ModernRonin** [sharpliner/sharpliner#168]
Hi,

maybe doing something wrong, but
```csharp
new PowershellFileTask("tools/scripts/RunDocker.ps1")
{
    Arguments = args,
    DisplayName = "Running docker image",
    Pwsh = true
};
```

creates this yaml:
```yaml
name: CreateReleasableImages

trigger:
  branches:
    include:
    - master
    - STABLE

variables:
- group: CommonPipelineVariables

- name: DOTNET_SKIP_FIRST_TIME_EXPERIENCE
  value: true

jobs:
- job: job
  pool:
    vmImage: ubuntu-latest
  steps:
  - checkout: self
    fetchDepth: 1

  - powershell: tools/scripts/RunDocker.ps1
    targetType: filepath
    arguments: -image ourcontainerregistry.azurecr.io/tools/pulumi:latest -command "/usr/bin/pwsh -Command {./vso.ps1; ./LogIntoAzure.ps1; ./RunPulumi.ps1 }" -doEnableDocker $true -$volumeMounts $(Build.SourcesDirectory)/image-maker:~/project
    displayName: Running docker image
    pwsh: true
```

which DevOps doesn't like:
>/tools/Pipelines/CreateReleasableImages.yml (Line: 31, Col: 5): Unexpected value 'targetType'
/tools/Pipelines/CreateReleasableImages.yml (Line: 32, Col: 5): Unexpected value 'arguments'
/tools/Pipelines/CreateReleasableImages.yml (Line: 34, Col: 5): Unexpected value 'pwsh'

Looking at the [documentation of the task](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/powershell?view=azure-devops), I get the impression that `powershell` is only for inline scripts and if one wants to use a file, one has to use the `PowerShell@2` task.

**Comment by ModernRonin** [comment]
Yes, that's definitely it. With 

```csharp
new AzureDevOpsTask("PowerShell@2")
            {
                Inputs = new TaskInputs
                {
                    ["targetType"] = "filepath",
                    ["filePath"] = "tools/scripts/RunDocker.ps1",
                    ["arguments"] = args,
                    ["pwsh"] = true
                },
                DisplayName = "Running docker image"
            };
```

it works.

**Comment by premun** [comment]
This is so weird - this documentation page says something completely else:
https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/powershell?view=azure-devops#arguments

I will have a look

**Comment by ModernRonin** [comment]
If you look at the section where they say "The PowerShell task also has two shortcuts in YAML:", the yaml examples for the shortcuts show a lot less parameters than the full powershell@ task.

So I guess for the shortcuts only those are available. In any case, they could be a LOT more explicit about these things.

Cheers,
MR

**Comment by premun** [comment]
Yeah, the docs are not great to be honest. But it makes Sharpliner useful at least :)

I will try to fix this soon (next week)

**Comment by ModernRonin** [comment]
Sharpliner is *very* useful!

btw, I opened another discussion...

**Comment by ModernRonin** [comment]
and the docs are catastrophic ;P

**Comment by premun** [comment]
Ah, I am getting emails about discussions now!

**Comment by erstert3st** [comment]
The same problem is still present in Pwsh.Inline("Write-Host 'I am not working'")

**Comment by premun** [comment]
This has been fixed in the version `1.5.17`.

**Comment by erstert3st** [comment]
thx :D
